x-pack/plugins/observability_solution/observability/server/lib/rules/custom_threshold/types.ts

```diff
@@ -46,7 +46,7 @@ export interface CustomThresholdRuleTypeParams extends RuleTypeParams {
 
 export type CustomThresholdRuleTypeState = RuleTypeState & {
   lastRunTimestamp?: number;
-  missingGroups?: Array<string | MissingGroupsRecord>;
```

Comment:
@simianhacker pointed out that string was used originally in the metric threshold and they changed the structure to `MissingGroupsRecord` and added `convertStringsToMissingGroupsRecord` for migration. We don't need this logic in the custom threshold as from the start, we were using `MissingGroupsRecord` structure.